src/accel/ExceptionConverter.cc

```diff
@@ -64,6 +74,67 @@ std::string strip_source_dir(std::string const& filename)
     return filename.substr(max_pos + 1);
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Write a detailed state message.
+ */
+void log_state(Logger::Message& msg,
+               KernelContextException const& kce,
+               SharedParams const* params)
+{
+    auto core_params = [params]() -> CoreParams const* {
+        if (!params)
+            return nullptr;
+        if (!*params)
+            return nullptr;
+        return params->Params().get();
+    }();
+
+    if (core_params && kce.particle())
+    {
+        auto const& par_params = *core_params->particle();
+        msg << "\n- Particle type: " << par_params.id_to_label(kce.particle())
+            << " (PDG=" << par_params.id_to_pdg(kce.particle()).get()
+            << ",  ID=" << kce.particle() << ')';
+    }
+    else
+    {
+        msg << "\n- Particle type ID: " << kce.particle();
+    }
+    msg << "\n- Energy: " << kce.energy() << "\n- Position: " << kce.pos()
+        << " (cm)"
+        << "\n- Direction: " << kce.dir();
```

Comment:
Can we add which physics process trigger the exception?

Reply:
Yes the active kernel, which is output, should give the most accurate picture. I could also add the post step action ID but that would require further changes to the KCE so perhaps a later pr?